vector-stores/spring-ai-cassandra-store/src/main/java/org/springframework/ai/vectorstore/cassandra/CassandraVectorStore.java

```diff
@@ -0,0 +1,1038 @@
+				alterTable = alterTable.addColumn(metadata.name(), metadata.type());
+			}
+			if (addContent) {
+				alterTable = alterTable.addColumn(this.schema.content, DataTypes.TEXT);
+			}
+			if (addEmbedding) {
+				// special case for embedding column, bc JAVA-3118, as above
+				StringBuilder alterTableStmt = new StringBuilder(((BuildableQuery) alterTable).asCql());
+				if (newColumns.isEmpty() && !addContent) {
+					alterTableStmt.append(" ADD (");
+				}
+				else {
+					alterTableStmt.setLength(alterTableStmt.length() - 1);
+					alterTableStmt.append(',');
+				}
+				alterTableStmt.append(this.schema.embedding)
+					.append(" vector<float,")
+					.append(vectorDimension)
+					.append(">)");
+
+				logger.debug("Executing {}", alterTableStmt.toString());
+				this.session.execute(alterTableStmt.toString());
+			}
+			else {
+				SimpleStatement stmt = ((AlterTableAddColumnEnd) alterTable).build();
+				logger.debug("Executing {}", stmt.getQuery());
+				this.session.execute(stmt);
+			}
+		}
+	}
+
+	/**
+	 * Indexes are automatically created with COSINE. This can be changed manually via
+	 * cqlsh
+	 */
+	public enum Similarity {
+
+		COSINE, DOT_PRODUCT, EUCLIDEAN
+
+	}
+
+	public enum SchemaColumnTags {
+
+		INDEXED
+
+	}
+
+	/**
+	 * Given a string document id, return the value for each primary key column.
+	 *
+	 * It is a requirement that an empty {@code List<Object>} returns an example formatted
+	 * id
+	 */
+	public interface DocumentIdTranslator extends Function<String, List<Object>> {
+
+	}
+
+	/** Given a list of primary key column values, return the document id. */
+	public interface PrimaryKeyTranslator extends Function<List<Object>, String> {
+
+	}
+
+	record Schema(String keyspace, String table, List<SchemaColumn> partitionKeys, List<SchemaColumn> clusteringKeys,
+			String content, String embedding, String index, Set<SchemaColumn> metadataColumns) {
+
+	}
+
+	public record SchemaColumn(String name, DataType type, SchemaColumnTags... tags) {
+
+		public SchemaColumn(String name, DataType type) {
+			this(name, type, new SchemaColumnTags[0]);
+		}
+
+		public GenericType<Object> javaType() {
+			return CodecRegistry.DEFAULT.codecFor(this.type).getJavaType();
+		}
+
+		public boolean indexed() {
+			for (SchemaColumnTags t : this.tags) {
+				if (SchemaColumnTags.INDEXED == t) {
+					return true;
+				}
+			}
+			return false;
+		}
+
+	}
+
+	/**
+	 * Builder for the Cassandra vector store.
+	 *
+	 * All metadata columns configured to the store will be fetched and added to all
+	 * queried documents.
+	 *
+	 * To filter expression search against a metadata column configure it with
+	 * SchemaColumnTags.INDEXED
+	 *
+	 * The Cassandra Java Driver is configured via the application.conf resource found in
+	 * the classpath. See
+	 * https://github.com/apache/cassandra-java-driver/tree/4.x/manual/core/configuration
+	 *
+	 */
+	public static class CassandraBuilder extends AbstractVectorStoreBuilder<CassandraBuilder> {
```

Comment:
does `CassandraBuilder` have to be an inner class ? 
no strong opinion, but i gut instinct is it's big enough to warrant being a separate java file.

Reply:
We can consider that - but wanted to make the usage consistent across all the vector stores.

Reply:
i would suggest that, given the lines of code and that the builder has a little more logic in it the a super simple builder.  (i had previously intentionally kept the classes separate for the sake of encapsulation between startup/configuration versus runtime behaviour).

it's not a hill i will die on.  if you take past PoVs into account, then i'm happy it's your call to make.